src/spec-node/devContainersSpecCLI.ts

```diff
@@ -134,14 +153,12 @@ function provisionOptions(y: Argv) {
 	})
 		.check(argv => {
 			const idLabels = (argv['id-label'] && (Array.isArray(argv['id-label']) ? argv['id-label'] : [argv['id-label']])) as string[] | undefined;
+			const isWorkspaceFound = argv['workspace-folder'] || findWorkspaceFolder();
 			if (idLabels?.some(idLabel => !/.+=.+/.test(idLabel))) {
 				throw new Error('Unmatched argument format: id-label must match <name>=<value>');
 			}
-			if (!(argv['workspace-folder'] || argv['id-label'])) {
-				throw new Error('Missing required argument: workspace-folder or id-label');
-			}
-			if (!(argv['workspace-folder'] || argv['override-config'])) {
-				throw new Error('Missing required argument: workspace-folder or override-config');
+			if (!(isWorkspaceFound || argv['id-label'] || argv['override-config'])) {
+				throw new Error('Missing required argument: workspace-folder, id-label, or override-config');
```

Comment:
As this PR automatically computes `workspace-folder`, I wonder if we need to remove `workspace-folder` from the `requires args` list 🤔 

If so, we'd need to update the error message accordingly.